### PR TITLE
docs: Jsdocs documentation

### LINKED ISSUE #6: Improve README
Add documentation on all methods and all options for those methods (and default values) and contributing guide

### PR DESCRIPTION
**Pull request by marcelorl**
It should close [#6](https://github.com/joshghent/query-stringifier/issues/6)

it includes:

 - Jsdocs documentation for each method
 - New npm script `npm run docs` which generates a new documentation within /docs folder
 - Contributing guidec

Please, in order to enable Github pages, just set `master branch /docs folder` in the repo Settings -> Git Pages section.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -41,8 +47,11 @@ QueryString.prototype.stringify = function (params, options) {
     return options.prefix + queryStringArray.join('&');
 };
 
-// Arguments:
-// @params - STRING - The query string to parse into an object
+/**
+ * Add the query stringify method
+ * @param {string | any} queryStr - The query string to parse into an object. If any type other than string, just returns it
+ * @returs {object} Returns query-string as object
```
Comment (joshghent) [inline index.js]: Looks like this is misspelled. Please correct to `returns` :)

**Comment by joshghent** [comment]
Awesome job @marcelorl - this is great. And will allow us to expand the docs with ease.
I wonder, do you know if there is a way to have the docs build automatically and pushed to the master branch?

File: index.js (hunk 1 above)
Comment (marcelorl) [reply index.js]: good one... fixed

**Comment by marcelorl** [comment]
Thank you @joshghent 

Yes, I think a githook pre-commit (achieved with husky) would solve it. what do you think?

Then, whenever anyone commit anything, docs would be generated and added to that commit just created.

**Comment by joshghent** [comment]
@marcelorl Yeah that might be a good shout! Raise a separate issue for that. This is great work 👍

**Comment by marcelorl** [comment]
sure....Im opening this new issue.

Thank you.

**Comment by marcelorl** [comment]
dont forget to add it to your github pages.

then, you get something like this:
https://marcelorl.github.io/query-stringifier/

haha

**Comment by joshghent** [comment]
@marcelorl It's up -> https://joshghent.github.io/query-stringifier/
Thanks dude, looks great
Feel free to add this to the readme

**Pull request merged**